### DISCUSSION TITLE
BCF viewpoint in not working in viewer

### DISCUSSION THREAD

**Discussion by Morpii** [IfcOpenShell/IfcOpenShell#5549] [Q&A]
I am using the ifctester library to validate an IFC file against an IDS, generate a report, and create a BCF file for any failed checks. However, I'm encountering an issue with the assigned viewpoints in the BCF when viewing them in Autodesk Construction Cloud (ACC).

The problems I’m facing are: while the issues are correctly located on the right object, when I click an issue, the viewpoint moves to a location far away from the model. It doesn't zoom in on or focus on the object connected to the issue.

I'm unsure if this is an issue with my implementation or something I’ve misunderstood. My expectation was that the viewpoint would automatically zoom in or correspond to the object associated with the issue.

I can see in the **ifctester.reporter.bcf.to_file()** method that a viewpoint is added to the BCF, and my code confirms that they are being included. However, I'm uncertain whether they are accurate, which is strange considering the issue is located correctly.

I also recognize this might be a viewer-specific issue, so my question is: how does the viewer decide on assigning the viewpoint? Given that there can be countless viewpoints for a single object (different angles, distances, etc.), what factors determine how the viewpoint is set?

**Comment by Morpii** [comment]
Update: 
I have extracted the viewpoint files from an issue from the ifctester.reporter.bcf.to_file() and an issue created in Autodesk Construction Cloud (ACC) . Where there is a differende in how the coordinates are written in different formats. Is there a way to overwrite the NP.FLOAT64 format?

Viewpoint file from ifctester.reporter.bcf.to_file():
![image](https://github.com/user-attachments/assets/772ac373-0cb0-40f1-af58-422b4437331f)

Viewpoint file from issue created in ACC:
![image](https://github.com/user-attachments/assets/807e8082-4140-480a-a308-c91755081a57)

**Comment by mariamsargsyan201** [comment]
Hello! I’m having the same issue as you described. Did you manage to find a solution? If so, I’d really appreciate it if you could share it with me.

**Reply by Morpii** [reply]
Shouldt have answered here, but ended ut as a new comment under :)

**Comment by Morpii** [comment]
Hi! I solved it by using regex substitution and rewriting it to the file. The "file_path" parameter is to the .bcfv file which is inside the .bcf file.
Hope this helps :)

`
   
    import xml.etree.ElementTree as ET
    import re

    def replace_npfloat64_in_bcfv(file_path):
    
        # Read the file content
        with open(file_path, "r", encoding="utf-8") as file:
            content = file.read()
    
        # Use regex to find and replace NP.FLOAT64(x) with just x
        content = re.sub(r"NP\.FLOAT64\(([^)]+)\)", r"\1", content)
    
        # Parse the modified content back to XML to ensure it's valid
        try:
            root = ET.fromstring(content)
            # Write the modified XML content back to the file
            with open(file_path, "w", encoding="utf-8") as file:
                file.write(ET.tostring(root, encoding='unicode'))
            # print(f"Updated file: {file_path}")
        except ET.ParseError as e:
            print(f"Error parsing the modified XML: {e}")
`

**Comment by mariamsargsyan201** [comment]
Thank you)
